src/Wyam/Settings.cs

```diff
@@ -49,6 +50,7 @@ public bool ParseArgs(string[] args, out bool hasErrors)
                 syntax.DefineOptionList("i|input", ref InputPaths, DirectoryPath.FromString, "The path(s) of input files, can be absolute or relative to the current folder.");
                 syntax.DefineOption("o|output", ref OutputPath, DirectoryPath.FromString, "The path to output files, can be absolute or relative to the current folder.");
                 syntax.DefineOption("c|config", ref ConfigFilePath, FilePath.FromString, "Configuration file (by default, config.wyam is used).");
+                syntax.DefineOption("self-update", ref SelfUpdate, "Tries to find and install an updated version of the application.");
```

Comment:
I love that updating will be as easy as just adding a CLI flag. The System.CommandLine library I'm using supports the concept of [*commands*](https://github.com/dotnet/corefxlab/tree/master/src/System.CommandLine#defining-commands), or different operating modes for the application. Perhaps the update process should be it's own command since you probably wouldn't want to both update and run a build at the same time. That would also allow us to eventually pass update-specific flags like version (don't know if Squirrel supports this), source, etc.

Hopefully System.CommandLine supports a default command and the current set of options can still be provided as such - if I have to differentiate between the update command and the build command on every execution, I'd rather keep it as a flag like it is here.